ABOUT-LICENSING.md

```diff
@@ -220,7 +220,25 @@ under the problematic JSON license):
 > OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 > SOFTWARE.
 
+
+For generating the documentation of Ginkgo, some scripts from the deal.II
+library are used. You can refer to the `doc/` folder to see which files are a
+modified version of deal.II's documentation generation scripts. Additionally,
+an example of interfacing Ginkgo with an external library is available in the
+`examples/` folder of Ginkgo. This uses the `example-9` of deal.II which is
+also licensed the same as the deal.II library.
+
+
+>The deal.II library is free software; you can use it, redistribute it, and/or
+>modify it under the terms of the GNU Lesser General Public License as published
+>by the Free Software Foundation; either version 2.1 of the License, or (at your
+>option) any later version. The full text of the GNU Lesser General Public
+>version 2.1 is available on the [deal.II repository
+>page](https://github.com/dealii/dealii/blob/master/LICENSE.md) or on the
+>official [GNU license
+>page](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html).
```

Comment:
Previous licenses have a space after the `>` sign, so for uniformity, I would recommend doing the same.
Additionally, the previous licenses also mention the header for the copyright:
```
 * Copyright (C) 2000 - 2018 by the deal.II authors
```
While reading a bit more into it, I am not sure if we are allowed to use their code, because Ginkgo is licensed under BSD 3-Clause, which is incompatible with their GLP 2.1.
Their license states:
>  For example, if you distribute copies of the library, whether gratis
or for a fee, you must give the recipients all the rights that we gave
you.  You must make sure that they, too, receive or can get the source
code.  If you link other code with the library, you must provide
complete object files to the recipients, so that they can relink them
with the library after making changes to the library and recompiling
it.  And you must show them these terms so they know their rights.

So I think we are only allowed to use their code if we publish the finished work under (L)GPL.
Here is a [stackexchange question about this problem](https://opensource.stackexchange.com/questions/35/if-i-include-some-gpl-code-in-my-project-can-i-release-it-as-bsd)

Reply:
LGPL is very different from GPL. LGPL was made so that it does not contaminate all projects unlike GPL. AFAIK deal.II uses LGPL. All you need is to make the modifications to a program under LGPL license. But if you reuse a part of the program (in a sense, link against it) you don't have to put all your code under LGPL also, that would be the case with GPL.

BTW, Pratik it is mandatory to remove our license from the deal-II example due to that.

https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License

Reply:
To make it even clearer, you cannot sub-license with LGPL.

Reply:
Oh, my bad, I thought there was only a slight difference between them. I guess I still have to read a bit more into that, thanks for the clarification!

Reply:
Hm, I see. I will rename the file to .cc then, so that we dont automatically add license to it and remove our license. But I think I can keep a note saying that we have modified it.

Reply:
Yes, the note that we modified it might actually be mandatory, but in any case it is not wrong to have it there.